### PR TITLE
[Data Plane Mgr] UT Test Automation - Part III

### PR DESCRIPTION
**Pull request by kevin-zhonghao**
The third version of DPM Automation UT:
1. Refactored the old version of UTs
2. Optimized the logic of automatically generating input and output payload
3. The verification content of UTs has been changed. The output protobuff was only roughly compared before, and now it can be specific to the value of each line

### CODE REVIEW

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/GoalStateManager.java [hunk 1]
```diff
@@ -46,418 +46,418 @@
...
-    Map<String, List<InternalPortEntity>> mapGroupedByHostIp = new HashMap();
-    Map<String, InternalSubnetEntity> subnetMap = new HashMap<>();
-    Map<String, InternalPortEntity> portMap = new HashMap<>();
-    Map<String, com.futurewei.alcor.web.entity.vpc.VpcEntity> vpcMap = new HashMap<>();
-    // construct map from list
-    for (InternalSubnetEntity s : subnetArr) {
-      subnetMap.put(s.getId(), s);
+    private static final Logger LOG = LoggerFactory.getLogger();
+
+    private void printNetworkConfiguration(NetworkConfiguration networkConfiguration)
```
Comment (xieus) [inline services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/GoalStateManager.java:51]: I think we could revert the change as it is related to PR #418 .

File: services/data_plane_manager/src/test/java/com/futurewei/alcor/dataplane/utils/DataPlaneManagerUtil.java [hunk 2]
```diff
@@ -355,31 +416,38 @@ private InternalSubnetEntity getInternalSubnetEntity(NetworkConfiguration networ
...
+//        List<String> subnetIdsInFixedIP = new ArrayList<>();
+//        for (Port.PortState portState: portStates) {
+//            for (Port.PortConfiguration.FixedIp fixedIp: portState.getConfiguration().getFixedIpsList()) {
+//                if (!subnetIdsInFixedIP.contains(fixedIp.getSubnetId())) {
+//                    InternalSubnetEntity internalSubnetEntity = getInternalSubnetEntity(
+//                            networkConfig, fixedIp.getSubnetId());
+//                    subnetEntities.add(internalSubnetEntity);
+//                    subnetIdsInFixedIP.add(fixedIp.getSubnetId());
+//                }
+//            }
```
Comment (xieus) [inline services/data_plane_manager/src/test/java/com/futurewei/alcor/dataplane/utils/DataPlaneManagerUtil.java:438]: Let us remove if not needed.

File: services/data_plane_manager/src/test/java/com/futurewei/alcor/dataplane/utils/DataPlaneManagerUtil.java [hunk 3]
```diff
@@ -79,21 +120,35 @@ public NetworkConfiguration autoGenerateUTsInput(int operationType,
             InternalRouterInfo routerInfo = new InternalRouterInfo();
             InternalRouterConfiguration internalRouterConfiguration = new InternalRouterConfiguration();
             List<InternalSubnetRoutingTable> subnet_routing_tables = new ArrayList<>();
+            for (int i = 0; i < subnetNum; i ++) {
```
Comment (xieus) [inline services/data_plane_manager/src/test/java/com/futurewei/alcor/dataplane/utils/DataPlaneManagerUtil.java:123]: @kevin-zhonghao As this change is a pretty big change, could you please generate a few input and output using the UT framework? Let us review them together tomorrow.

**Comment by xieus** [comment]
Linked to Issue #409.

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/GoalStateManager.java (hunk 1 above)
Comment (kevin-zhonghao) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/GoalStateManager.java:51]: Sure

File: services/data_plane_manager/src/test/java/com/futurewei/alcor/dataplane/utils/DataPlaneManagerUtil.java (hunk 2 above)
Comment (kevin-zhonghao) [reply services/data_plane_manager/src/test/java/com/futurewei/alcor/dataplane/utils/DataPlaneManagerUtil.java:438]: Sure

File: services/data_plane_manager/src/test/java/com/futurewei/alcor/dataplane/utils/DataPlaneManagerUtil.java (hunk 3 above)
Comment (kevin-zhonghao) [reply services/data_plane_manager/src/test/java/com/futurewei/alcor/dataplane/utils/DataPlaneManagerUtil.java:123]: NP

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/GoalStateManager.java (hunk 1 above)
Comment (xieus) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/GoalStateManager.java:51]: > I think we could revert the change as it is related to PR #418 .

@kevin-zhonghao I think this PR still contains payload from PR #418, which should not be included in this PR.

**Review by xieus — CHANGES_REQUESTED** [review]

**Comment by kevin-zhonghao** [comment]
Solved reverted the GoalStateManager.class to alcor master version

**Review by xieus — APPROVED** [review]
LGTM.
